internalshared/configutil/encrypt_decrypt_test.go

```diff
@@ -0,0 +1,82 @@
+package configutil
+
+import (
+	"context"
+	"testing"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+)
+
+func getAEADTestKMS(t *testing.T) {
+}
+
+func TestEncryptParams(t *testing.T) {
+	rawStr := `
+storage "consul" {
+	api_key = "{{encrypt(foobar)}}"
+}
+
+telemetry {
+	some_param = "something"
+	circonus_api_key = "{{encrypt(barfoo)}}"
+}
+`
+
+	finalStr := `
+storage "consul" {
+	api_key = "foobar"
+}
+
+telemetry {
+	some_param = "something"
+	circonus_api_key = "barfoo"
+}
+`
+
+	reverser := new(reversingWrapper)
+	out, err := EncryptDecrypt(rawStr, false, false, reverser)
+	if err != nil {
+		t.Fatal(err)
+	}
+
```

Comment:
Should you check the encrypted output contains `{{decrypt(oofrab)}}`?

Reply:
I actually was thinking about that -- technically it contains a marshaled proto but I can just unmarshal in the test and verify. I sort of figured so long as what came in == what goes out it doesn't truly matter what it looks like in the middle 😂  But I'll add it.